cel/folding.go

```diff
@@ -139,6 +145,32 @@ func tryFold(ctx *OptimizerContext, a *ast.AST, expr ast.Expr) error {
 	return nil
 }
 
+func getFunctionOverloadDecls(function *decls.FunctionDecl, overloadIds []string) []*decls.OverloadDecl {
+	var overloads []*decls.OverloadDecl
+	if function == nil {
+		return overloads
+	}
+	for _, o := range function.OverloadDecls() {
+		for _, id := range overloadIds {
+			if id == o.ID() {
+				overloads = append(overloads, o)
+			}
+		}
+	}
+	return overloads
+}
+
+func isLateBoundFunctionCall(ctx *OptimizerContext, a *ast.AST, expr ast.Expr) bool {
+	call := expr.AsCall()
+	overloadDecls := getFunctionOverloadDecls(ctx.Functions()[call.FunctionName()], a.GetOverloadIDs(expr.ID()))
```

Comment:
Consider iterating over the functions and overloads inline since you just need to check for existence rather than allocate the overload subset.

Reply:
Alternatively, you could expose `HasLateBinding()` at the `decls.FunctionDecl` level since this will be useful in generating an `Activation` with slots carved out for associating the function implementation with the activation instance.

Reply:
I added `HasLateBinding()` to `decls.FunctionDecl`. Since we enforce that functions cannot mix late and non-late bindings, we also don't need to iterate over the overloads here. This greatly simplifies the code.